packages/frontend/web/lib/raven.ts

```diff
@@ -0,0 +1,82 @@
+import raven, { RavenStatic } from 'raven-js';
+import { get } from '@frontend/web/lib/appData';
+
+let ravenConfig: RavenStatic;
+
+export const getRaven: () => RavenStatic = () => {
+    if (ravenConfig) {
+        return ravenConfig;
+    }
+
+    const { sentryPublicApiKey, sentryHost, isDev, switches } = get('config');
+    const { editionLongForm, contentType } = get('CAPI');
+
+    const sentryUrl = `https://${sentryPublicApiKey}@${sentryHost}`;
+    const sentryOptions = {
+        whitelistUrls: [
+            // localhost will not log errors, but call `shouldSendCallback`
+            /webpack-internal/,
+            /localhost/,
+            /assets\.guim\.co\.uk/,
+            /ophan\.co\.uk/,
+        ],
+        tags: {
+            contentType,
+            edition: editionLongForm,
+        },
+        ignoreErrors: [
+            "Can't execute code from a freed script",
+            /There is no space left matching rules from/gi,
+            'Top comments failed to load:',
+            'Comments failed to load:',
+            /InvalidStateError/gi,
+            /Fetch error:/gi,
+            'Network request failed',
+            'This video is no longer available.',
+            'UnknownError',
+            'Fetch error while requesting https://api.nextgen.guardianapps.co.uk/weatherapi/city.json:',
```

Comment:
We can probably remove a few of the frontend-specific ones:

- There is no space left matching rules from
- Top comments failed to load
- Comments failed to load
- Fetch error while requesting https://api.nextgen.guardianapps.co.uk/weatherapi/city.json

Others are questionable, but let's keep them for now